### PR TITLE
update to argument parser 1.2.2

### PR DESCRIPTION
**Pull request by tomerd**
update argument parser to version 1.2.0

### CODE REVIEW

**Comment by tomerd** [comment]
cc @natecook1000 @MaxDesiatov

**Comment by MaxDesiatov** [comment]
This version won't work with SwiftPM, we need one that includes https://github.com/apple/swift-argument-parser/pull/548, which hasn't been tagged yet AFAIU

**Comment by tomerd** [comment]
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci test

**Comment by tomerd** [comment]
> This version won't work with SwiftPM, we need one that includes https://github.com/apple/swift-argument-parser/pull/548, which hasn't been tagged yet AFAIU

in what way? it builds okay afaict

**Comment by MaxDesiatov** [comment]
> in what way? it builds okay afaict

It may build using SwiftPM, but the bootstrap script will fail due to a regression that was introduced in 1.2.0 and that https://github.com/apple/swift-argument-parser/pull/548 fixes.

**Comment by tomerd** [comment]
> It may build using SwiftPM, but the bootstrap script will fail due to a regression that was introduced in 1.2.0 and that https://github.com/apple/swift-argument-parser/pull/548 fixes.

@natecook1000 do we need a tag then?

**Comment by natecook1000** [comment]
@tomerd Just tagged 1.2.2, which has the fix that @MaxDesiatov mentioned 👍🏻

**Comment by tomerd** [comment]
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci test

File: utils/update_checkout/update-checkout-config.json [hunk 1]
```diff
@@ -96,7 +96,7 @@
                 "llbuild": "main",
                 "swift-tools-support-core": "main",
                 "swiftpm": "main",
-                "swift-argument-parser": "1.1.4",
+                "swift-argument-parser": "1.2.0",
```
Comment (tomerd) [inline utils/update_checkout/update-checkout-config.json:99]:
```suggestion
                "swift-argument-parser": "1.2.2",
```

File: utils/update_checkout/update-checkout-config.json [hunk 2]
```diff
@@ -423,7 +423,7 @@
...
```
Comment (tomerd) [inline utils/update_checkout/update-checkout-config.json:426]:
```suggestion
                "swift-argument-parser": "1.2.2",
```

**Comment by tomerd** [comment]
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci test

**Comment by tomerd** [comment]
```
Failed Tests (1):
  Swift(iphonesimulator-x86_64) :: SourceKit/CursorInfo/static_vs_class_spelling.swift
```

seems unrelated

**Comment by tomerd** [comment]
apple/swift-driver#1285
apple/swift-package-manager#6131
apple/sourcekit-lsp#708

@swift-ci test

**Comment by tomerd** [comment]
@compnerd do we need to fix anything for the windows build to pass?

**Comment by compnerd** [comment]
This seems like something else?

```
LLVM ERROR: out of memory

Allocation failed
```

**Comment by compnerd** [comment]
@swift-ci please build toolchain Windows platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci please build toolchain windows platform

**Comment by compnerd** [comment]
Please test with following PRs:
https://github.com/apple/swift-driver/pull/1285
https://github.com/apple/swift-package-manager/pull/6131
https://github.com/apple/sourcekit-lsp/pull/708

@swift-ci please test Windows platform

**Comment by shahmishal** [comment]
Updated the title to match the tag version (1.2.2)

**Pull request merged**